x-pack/plugins/apm/server/lib/metrics/types.ts

```diff
@@ -4,37 +4,43 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import {
+  PromiseReturnType,
+  IndexAsString,
+  Omit
+} from '../../../typings/common';
 import { ChartType, YUnit } from '../../../typings/timeseries';
+import { fetch as cpuFetch } from './by_agent/shared/cpu/fetcher';
 
-export interface AggValue {
-  value: number | null;
-}
+type CpuSearchResponse = PromiseReturnType<typeof cpuFetch>;
 
-export interface MetricSeriesKeys {
-  [key: string]: AggValue;
-}
+export type MetricSearchResponse<MetricNames extends string> = Omit<
+  CpuSearchResponse,
+  'aggregations'
+> & {
+  aggregations: IndexAsString<
```

Comment:
Sometimes TS needs a little help in realising object keys are strings (something to do with the keyof operator).